### PR TITLE
FS HTTP API ACL enforcement

### PR DESCRIPTION
**Pull request by schmichael**
ACL enforcement for the filesystem HTTP APIs on clients.

### CODE REVIEW

File: command/agent/fs_endpoint.go [hunk 1]
```diff
@@ -70,19 +71,44 @@ func (s *HTTPServer) FsRequest(resp http.ResponseWriter, req *http.Request) (int
...
+		return requireReadFS(s.FileReadAtRequest)
 	case strings.HasPrefix(path, "cat/"):
-		return s.FileCatRequest(resp, req)
+		return requireReadFS(s.FileCatRequest)
 	case strings.HasPrefix(path, "stream/"):
-		return s.Stream(resp, req)
+		return requireReadFS(s.Stream)
 	case strings.HasPrefix(path, "logs/"):
+		// Logs can be accessed with ReadFS or ReadLogs caps
+		if aclObj, err := s.agent.Client().ResolveToken(secret); err != nil {
```
Comment (dadgar) [inline command/agent/fs_endpoint.go]: Can you just resolve the token and check the error at line 79 and then the helper func and this just do the else if part.

File: command/agent/fs_endpoint_test.go [hunk 2]
```diff
@@ -106,6 +109,124 @@ func TestAllocDirFS_ReadAt_MissingParams(t *testing.T) {
...
 }
 
+func TestAllocDirFS_ACL(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	httpACLTest(t, nil, func(s *TestAgent) {
+		state := s.Agent.server.State()
+
+		req, err := http.NewRequest("GET", "/v1/client/fs/ls/", nil)
```
Comment (dadgar) [inline command/agent/fs_endpoint_test.go]: If you can, could you just wrap this in a for loop that runs the same suite of tests against all endpoints:

```
for _, endpoint := range []string{"ls", "stat", ...} {
  go.Run(endpoint, func(t *testing.T) {
      req, err := http.NewRequest("GET", fmt.Sprintf("/v1/client/fs/%s/", endpoint), nil)
      ...
  })
}
```

File: website/source/api/client.html.md [hunk 3]
```diff
@@ -529,9 +529,9 @@ The table below shows this endpoint's support for
 [blocking queries](/api/index.html#blocking-queries) and
 [required ACLs](/api/index.html#acls).
 
-| Blocking Queries | ACL Required |
-| ---------------- | ------------ |
-| `NO`             | `none`       |
+| Blocking Queries | ACL Required        |
```
Comment (dadgar) [inline website/source/api/client.html.md]: Document the GC endpoint

Comment (schmichael) [reply website/source/api/client.html.md]: Added in https://github.com/hashicorp/nomad/pull/3343

**Pull request merged**